rgw/v2/tests/s3_swift/test_swift_static_large_object_expiration.py

```diff
@@ -0,0 +1,199 @@
+import v2.utils.utils as utils
+from v2.lib.admin import UserMgmt
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.resource_op import Config
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+import v2.lib.manage_data as manage_data
+from v2.lib.swift.auth import Auth
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+import time
+from datetime import datetime,timezone, timedelta
+import random, string
+import json
+from swiftclient import ClientException
+import concurrent.futures
+log = logging.getLogger()
+
+
+TEST_DATA_PATH = None
+
+
+
+def get_unique_name(length=7):
+    characters = string.ascii_letters + string.digits
+    return "".join(random.choices(characters,k=length))
+
+container_name = "swiftuser_"+get_unique_name()
+object_name = 'object_'+get_unique_name()
+segment_prefix = f"{object_name}/segment"
+segment_size_test =100000000
+filename_test= "a_large_file"+get_unique_name(3)
+
+def create_a_large_file(filename):
+    file_path = os.path.join(TEST_DATA_PATH, filename)
+    data_info = manage_data.io_generator(file_path, 1073741824) #1GB File Creation
+    # Container and object detail
+    print(f"DATA INFO :: {data_info}")
+
+def upload_segments(rgw,filename,segment_size):
+    """Upload segments of the binary file."""
+    segment_list = []
+    file_path = os.path.join(TEST_DATA_PATH, filename)
+
+    with open(file_path, 'rb') as f:
+        segment_number = 0
+        while True:
+            segment_data = f.read(segment_size)
+            if not segment_data:
+                break
+
+            # Segment name
+            segment_name = f"{object_name}/segment_{segment_number:08d}"
+
+            # Upload the segment
+            rgw.put_object(container_name, segment_name, contents=segment_data)
+
+            # Add segment metadata to the manifest list
+            segment_list.append({
+                'path': f"/{container_name}/{segment_name}",
+                'etag': rgw.head_object(container_name, segment_name)['etag'],
+                'size_bytes': len(segment_data)
+            })
+
+            print(f"Uploaded segment: {segment_name}")
+            segment_number += 1
+
+    return segment_list
+
+def create_slo_manifest(rgw, segment_list):
+    """Create the manifest file for the SLO."""
+    headers = {'X-Static-Large-Object': 'True'}
+    rgw.put_object(container_name, object_name, contents=json.dumps(segment_list), headers=headers)
+    print(f"SLO manifest created for '{object_name}'.")
+
+
+def set_expiration(rgw):
+    """Sets expiration for the SLO."""
+    expiration_time = (datetime.now(timezone.utc) + timedelta(seconds=60)).strftime('%s')  # Expire in 60 seconds
+    rgw.post_object(container_name, object_name, headers={'X-Delete-At': expiration_time})
+    print(f"SLO '{object_name}' will expire at {datetime.fromtimestamp(int(expiration_time), tz=timezone.utc)} UTC.")
+
+def verify_expiration(rgw):
```

Comment:
IMO, most of the def functions other than test_exec, can be place in : ceph-qe-scripts/rgw/v2/tests/s3_swift/reusable.py for future use